src/third-version.ts

```diff
@@ -0,0 +1,524 @@
+  const components = asyncapi.components;
+  if (!isPlainObject(components)) {
+    return;
+  }
+
+  if (options.convertServerComponents && isPlainObject(components.servers)) {
+    components.servers = convertServerObjects(components.servers, asyncapi);
+  }
+  if (options.convertChannelComponents && isPlainObject(components.channels)) {
+    components.channels = convertChannelObjects(components.channels, asyncapi, options, context, true);
+  }
+  if (isPlainObject(components.securitySchemes)) {
+    components.securitySchemes = convertSecuritySchemes(components.securitySchemes);
+  }
+
+  if (isPlainObject(components.messages)) {
+    components.messages = convertMessages({
+      messages: components.messages
+    });
+  }
+}
+
+/**
+ * Convert `channels`, `servers` and `securitySchemes` in components.
+ */
+function convertSecuritySchemes(securitySchemes: Record<string, any>): Record<string, any> {
+  const newSecuritySchemes: Record<string, any> = {};
+  Object.entries(securitySchemes).forEach(([name, scheme]) => {
+    newSecuritySchemes[name] = convertSecuritySchemeObject(scheme);
+  });
+  return newSecuritySchemes;
+}
+
+/**
+ * Unify referencing mechanism in security field
+ */
+function convertSecurityObject(security: Array<Record<string, Array<string>>>, asyncapi: AsyncAPIDocument) {
+  const newSecurity: Array<any> = [];
+  security.forEach(securityItem => {
+    Object.entries(securityItem).forEach(([securityName, scopes]) => {
+      // without scopes - use ref
+      if (!scopes.length) {
+        newSecurity.push(createRefObject('components', 'securitySchemes', securityName))
+        return;
+      }
+
+      // create new security scheme in the components/securitySchemes with appropriate scopes
+      const securityScheme = getValueByPath(asyncapi, ['components', 'securitySchemes', securityName]);
+      // handle logic only on `oauth2` and `openIdConnect` security mechanism
+      if (securityScheme.type === 'oauth2' || securityScheme.type === 'openIdConnect') {
+        const newSecurityScheme = convertSecuritySchemeObject(securityScheme);
+        newSecurity.push({
+          ...newSecurityScheme,
+          scopes: [...scopes],
+        });
+      }
+    });
+  });
+  return newSecurity;
+}
+
+const flowKinds = ['implicit', 'password', 'clientCredentials', 'authorizationCode'];
+/**
+ * Convert security scheme object to new from v3 version (flow.[x].scopes -> flow.[x].availableScopes).
+ */
+function convertSecuritySchemeObject(original: any) {
+  const securityScheme = JSON.parse(JSON.stringify(original));
+  if (securityScheme.flows) {
+    flowKinds.forEach(flow => {
+      const flowScheme = securityScheme.flows[flow];
+      if (flowScheme && flowScheme.scopes) {
+        flowScheme.availableScopes = flowScheme.scopes;
+        delete flowScheme.scopes;
+      }
+    });
+  }
+  return securityScheme;
```

Comment:
@magicmatatjahu where does this change originate from? I cant find anything in the spec changes that suggest this change?

Reply:
https://github.com/asyncapi/spec/blob/next-major-spec/spec/asyncapi.md#fixed-fields-27 It was introduced by Fran

Reply:
Thanks!